From f63eef79f5cc1d76bd2d4955b16312bb4fe4cf00 Mon Sep 17 00:00:00 2001
From: Alexey Brodkin <Alexey.Brodkin@synopsys.com>
Date: Mon, 13 Mar 2017 20:41:45 +0300
Subject: [PATCH 1/2] toolchain: Disable PIE for ARC

Even though ARC's toolchain may build PIE binaries
those binaries ane not really functional in runtime as of today.
On ARC we don't expect text sections to be updated in run-time and
so when dynamic loader attempts to patch text section we're getting
access violation exception and application gets killed.

So until we improve support of PIE in run-time let's force-disable PIE
in Buildroot for ARC. Oherwise autotools will continue to detect PIE and
build OpenSSH and friends with PIE which leads to completely unexpected
"crashes" later on on the target.

Signed-off-by: Alexey Brodkin <abrodkin@synopsys.com>
Cc: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Cc: Peter Korsgaard <peter@korsgaard.com>
---
 toolchain/Config.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/toolchain/Config.in b/toolchain/Config.in
index de0158567..26b5e62f1 100644
--- a/toolchain/Config.in
+++ b/toolchain/Config.in
@@ -19,7 +19,7 @@ config BR2_TOOLCHAIN_USES_GLIBC
 
 config BR2_TOOLCHAIN_USES_UCLIBC
 	bool
-	select BR2_TOOLCHAIN_SUPPORTS_PIE if !BR2_m68k && !BR2_microblaze && !BR2_STATIC_LIBS
+	select BR2_TOOLCHAIN_SUPPORTS_PIE if !BR2_arc && !BR2_m68k && !BR2_microblaze && !BR2_STATIC_LIBS
 
 config BR2_TOOLCHAIN_USES_MUSL
 	bool
-- 
2.11.0
